### ISSUE TITLE
Buerokratt-Chatbot frontent bug-fix

### ISSUE THREAD

**Issue by varmoh** [buerokratt/Buerokratt-Chatbot#343]
Currently, cs-get-user-profile-settings.yml is intended to be a POST query- In frontend, its being sent to Ruuter as GET.

**Comment by varmoh** [comment]
@MarkoKull Please add your comment about this bug,  that we talked in a call.

**Comment by MarkoKull** [comment]
@turnerrainer please add your opinion.

Discussed this with @varmoh.
The reason this ticket is created, is because the router dsl script expects the request having body with userid value.
To change the request from GET to POST requires changing also all the components/modules where this endpoint is called.
It also requires to add some additional checks, that if the URL is 'cs-get-user-profile-settings', then call POST instead of GET.
That means (multiple) changes in every module.

What I suggest is, that we can update only router dsl script and leave the request method to be GET.
The request has cookie set, and that cookie contains personal code value.
Lets just add one additional step, something like get_user_id_from_cookie (that extracts personal code from cookie and assign it to userid variable). 
Userid I think is same as personal code, and even if it is not, we can add one more step find_userid_by_personalcode
That way we don't need any parameter or body.  Also security auditor is happy.

**Comment by turnerrainer** [comment]
> @turnerrainer please add your opinion.
> 
> Discussed this with @varmoh. The reason this ticket is created, is because the router dsl script expects the request having body with userid value. To change the request from GET to POST requires changing also all the components/modules where this endpoint is called. It also requires to add some additional checks, that if the URL is 'cs-get-user-profile-settings', then call POST instead of GET. That means (multiple) changes in every module.
> 
> What I suggest is, that we can update only router dsl script and leave the request method to be GET. The request has cookie set, and that cookie contains personal code value. Lets just add one additional step, something like get_user_id_from_cookie (that extracts personal code from cookie and assign it to userid variable). Userid I think is same as personal code, and even if it is not, we can add one more step find_userid_by_personalcode That way we don't need any parameter or body and we don't need to change any existing request. Also security auditor is happy.

@MarkoKull what do you mean by `Also security auditor is happy`? On the contrary, sending a personal code, user ID or any other significant variable used to make personal requests straight from the frontend to backend is a major security issue.

This Ruuter DSL was previously `GET` and is now `POST`. However, it should be turned back from `POST` to `GET`, but the DSL should get the `chatId` parameter from appropriate cookie passed on to Ruuter as a header. Will be an easy fix on Ruuter DSL side and no GUI developments are needed.

**Comment by 1AhmedYasser** [comment]
@turnerrainer i agree, user id should be extracted from the token not passed as a parameter for security and to avoid unnecessary parameters passed while i already have it 👌 , there are multiple endpoints who do that, its not just this one, we need to check all the requests and make sure that in all requests , user id is passed from the cookie 👌

**Comment by 1AhmedYasser** [comment]
@turnerrainer Will be moved to Blocked Status until changes are made to header by @ExiRain
